File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    one = [1,2,3,4,5]*(len(answers)//5+1)
    two = [2, 1, 2, 3, 2, 4, 2, 5] * (len(answers)//8+1)
    three = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5] * (len(answers)//10+1)
    ones = 0
    twos = 0
    threes = 0
    for i,n in enumerate(answers):
        if n == one[i]:
            ones+=1
        if n == two[i]:
            twos+=1
        if n == three[i]:
            threes+=1
    sc = [ones, twos, threes]
    ma = max(sc)
    for i in range(len(sc)):
        if sc[i] == ma:
            answer.append(i+1)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> n1 = {1,2,3,4,5};
    vector<int> n2 = {2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> n3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    vector<int> score(3);
    for(int i = 0; i < answers.size(); i++){
        if(answers[i] == n1[i%n1.size()])
            score[0]+=1;
        if(answers[i] == n2[i%n2.size()])
            score[1]+=1;
        if(answers[i] == n3[i%n3.size()])
            score[2]+=1;
    }
    
    int max = *max_element(score.begin(), score.end());
    for(int i = 0 ; i < 3; i++){
        if(max == score[i])
            answer.push_back(i+1);
    }
    return answer;
}